### ISSUE TITLE
RecHit printouts (IMPOSSIBLE, perfect match not found)

### ISSUE THREAD

**Issue by makortel** [cms-patatrack/cmssw#71]
I noticed from the log of "ttbar testing 10824.8" with TTbar+PU for #62 as the input the following printouts from the rechits
```
Begin processing the 28th record. Run 1, Event 31, LumiSection 1 on stream 0 at 31-May-2018 18:11:13.575 CEST
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:11:14 CEST Run: 1 Event: 31
perfect Match not found 1435/9851/0/9861 2583715!=2856745 0/0/0
%MSG
...
Begin processing the 66th record. Run 1, Event 115, LumiSection 3 on stream 0 at 31-May-2018 18:11:44.050 CEST
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:11:44 CEST Run: 1 Event: 115
IMPOSSIBLE 1334/12427/0/12428 2332265/2407919 6!=1
%MSG
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:11:44 CEST Run: 1 Event: 115
perfect Match not found 1334/12427/0/12428 2332265!=2407919 6/1/1
%MSG
...
Begin processing the 94th record. Run 1, Event 146, LumiSection 3 on stream 0 at 31-May-2018 18:12:06.270 CEST
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:12:06 CEST Run: 1 Event: 146
IMPOSSIBLE 1241/7716/9/7729 9098/3756146 83!=2
%MSG
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:12:06 CEST Run: 1 Event: 146
perfect Match not found 1241/7716/9/7729 9098!=3756146 83/2/2
%MSG
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:12:06 CEST Run: 1 Event: 146
IMPOSSIBLE 1241/7716/10/7741 8490/9098 86!=32769
%MSG
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:12:06 CEST Run: 1 Event: 146
IMPOSSIBLE 1241/7716/11/7740 2763678/8490 91!=31
%MSG
%MSG-w GPUHits2CPU:  SiPixelRecHitHeterogeneous:siPixelRecHitsPreSplitting  31-May-2018 18:12:06 CEST Run: 1 Event: 146
perfect Match not found 1241/7716/11/7740 2763678!=8490 91/31/31
%MSG
```
Link to full log
https://fwyzard.web.cern.ch/fwyzard/patatrack/pulls/f24fdf85aaaf7542195d985cacab4640b4ae5efb/RelValTTbar_13-CMSSW_10_2_0_pre3-PU25ns_101X_upgrade2018_realistic_v7-v1/testing-10824.8-step3.log

(the "development" log, i.e. without #62, shows the same printouts)

@VinInn Are the IMPOSSIBLE ones something that should be investigated at some point?

**Comment by cmsbot** [comment]
A new Issue was created by  @makortel Matti Kortelainen.

 can you please review it and eventually sign/assign? Thanks.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-issues.html">here</a>

**Comment by VinInn** [comment]
yes, this is due to the difference between cpu and gpu clustering: on cpu the size of a cluster is limited
while on gpu is not

**Comment by makortel** [comment]
We discussed with @VinInn, and these are something to be checked after we have the full workflow (on GPU) working.

**Comment by fwyzard** [comment]
since it looks like the GPU algorithm behaves "better" than the CPU one, would it make sense to "improve" it in the same way ?

**Comment by VinInn** [comment]
It will require a full involvement of Pixel DPG.
I suggested to "go public" with them once Run2 is finished and we hope they will have more time to dedicate to software for Run3.
(we can of course do it for the limited case of Phase1 pixel in run3 as a specific Raw2Cluster in a similar way I did it for Strip)
